### PR TITLE
FIX: Remove deprecated pkgversion; code reformats; fix 2.7 type hints error

### PR DESCRIPTION
**Pull request by aikikode**
A follow-up for https://github.com/kpn-digital/py-fqn-decorators/pull/17 because linting checks failed to 2.7 and pkgversion is deprecated and won't work with new pip versions. So for now the latest master can't be used.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,6 +6,8 @@
 DEPS:=requirements.txt
 DOCKER_COMPOSE:=$(shell which docker-compose)
 
+PYPI_UPLOAD_URL:="https://artifacts.kpn.org/api/pypi/pypi"
```
Comment (ricardosantosalves) [inline Makefile]: This is public package, we upload to public pypi.

Comment (ilhanmercan) [inline Makefile]: it's a public repo. not sure if this should be added here.

File: fqn_decorators/async.py [hunk 2]
```diff
@@ -1,9 +1,9 @@
+# flake8: noqa
 import warnings
 
 warnings.warn(
-    'This module was renamed to `asynchronous` due to adding `async`'
-    ' keyword in python 3.7',
-    PendingDeprecationWarning
+    "This module was renamed to `asynchronous` due to adding `async` keyword in python 3.7",
+    PendingDeprecationWarning,
```
Comment (kammala) [inline fqn_decorators/async.py]: should we drop this file instead?

as we are doing major bump anyway

File: tox.ini [hunk 3]
```diff
@@ -2,32 +2,37 @@
...
 
-[testenv:lint]
+[testenv:flake8]
 basepython = python3.5
-commands = flake8 decorators tests --exclude=fqn_decorators/__init__.py
+commands = flake8 fqn_decorators tests setup.py --exclude=fqn_decorators/__init__.py
 deps = flake8
 
+[testenv:black]
+basepython = python3.5
```
Comment (kammala) [inline tox.ini]: wow, isn't it too old?

File: requirements.txt [hunk 4]
```diff
@@ -1,3 +1,25 @@
...
+mypy
+isort[pyproject]
+flake8
+black
+
+# Tests
+tox
+# Working around the bug introduced in 20.0.0 which fails in-Docker builds
+# https://github.com/tox-dev/tox/issues/1516
+virtualenv<20.0.0
```
Comment (kammala) [inline requirements.txt]: I don't think this is needed anymore

File: requirements.txt [hunk 5]
```diff
@@ -1,3 +1,25 @@
-# Additional packages
--r requirements/requirements-base.txt
--r requirements/requirements-testing.txt
+# Package in develop mode
+-e .
+
+importlib-metadata<3
```
Comment (kammala) [inline requirements.txt]: would be nice to have an explanation for this version pin

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+[tool.black]
+line-length = 120
+target_version = ["py27", "py35"]
```
Comment (kammala) [inline pyproject.toml]: should we also include newer pythons like py36, py37?

**Review by ilhanmercan — DISMISSED** [review]

**Review by kammala — APPROVED** [review]

**Pull request merged**